src/main/java/org/opentripplanner/graph_builder/module/osm/tagmapping/PortlandMapper.java

```diff
@@ -0,0 +1,47 @@
+package org.opentripplanner.graph_builder.module.osm.tagmapping;
+
+import static org.opentripplanner.graph_builder.module.osm.WayPropertiesBuilder.withModes;
+import static org.opentripplanner.graph_builder.module.osm.specifier.ExactMatchSpecifier.exact;
+import static org.opentripplanner.street.model.StreetTraversalPermission.ALL;
+
+import org.opentripplanner.graph_builder.module.osm.WayPropertySet;
+import org.opentripplanner.graph_builder.module.osm.specifier.Condition.Absent;
+import org.opentripplanner.graph_builder.module.osm.specifier.Condition.GreaterThan;
+import org.opentripplanner.graph_builder.module.osm.specifier.ExactMatchSpecifier;
+
+public class PortlandMapper implements OsmTagMapper {
+
+  @Override
+  public void populateProperties(WayPropertySet props) {
+    props.setMixinProperties("footway=sidewalk", withModes(ALL).walkSafety(1.1));
+    props.setMixinProperties(
+      new ExactMatchSpecifier(new Absent("name")),
+      withModes(ALL).walkSafety(1.2)
+    );
+    props.setMixinProperties("highway=trunk", withModes(ALL).walkSafety(1.2));
+    props.setMixinProperties("highway=trunk_link", withModes(ALL).walkSafety(1.2));
+    props.setMixinProperties("highway=primary", withModes(ALL).walkSafety(1.2));
+    props.setMixinProperties("highway=primary_link", withModes(ALL).walkSafety(1.2));
+    props.setMixinProperties("highway=secondary", withModes(ALL).walkSafety(1.1));
+    props.setMixinProperties("highway=secondary_link", withModes(ALL).walkSafety(1.1));
+    props.setMixinProperties("highway=tertiary", withModes(ALL).walkSafety(1.1));
+    props.setMixinProperties("highway=tertiary_link", withModes(ALL).walkSafety(1.1));
+    props.setMixinProperties(
+      new ExactMatchSpecifier(new GreaterThan("lanes", 4)),
+      withModes(ALL).walkSafety(1.1)
+    );
+    props.setMixinProperties("sidewalk=both", withModes(ALL).walkSafety(0.8));
+    props.setMixinProperties("sidewalk=left", withModes(ALL).walkSafety(0.9));
+    props.setMixinProperties("sidewalk=right", withModes(ALL).walkSafety(0.9));
+    props.setMixinProperties("surface=unpaved", withModes(ALL).walkSafety(1.4));
+    // high penalty for streets with no sidewalk
+    props.setMixinProperties(exact("sidewalk=no;maxspeed=55 mph"), withModes(ALL).walkSafety(6));
```

Comment:
Could we add a comment explaining why `exact()` is needed here but not in the above rules? Could be confusing to someone seeing this for the first time.

Reply:
Done.